packages/block-library/src/template-part/edit/inner-blocks.js

```diff
@@ -36,7 +38,83 @@ function useLayout( layout ) {
 	}
 }
 
-export default function TemplatePartInnerBlocks( {
+const parsedBlocksCache = new WeakMap();
+
+function NonEditableTemplatePartPreview( {
+	postId: id,
+	layout,
+	tagName: TagName,
+	blockProps,
+} ) {
+	useBlockEditingMode( 'disabled' );
+
+	const { content, editedBlocks } = useSelect(
```

Comment:
Actually I missed something here. The difference should be that the `getEntityRecord` call here uses `context: 'view'` which the `useEntityBlockEditor` hook does not support

Reply:
Can we add a "readonly" option to the `useEntityBlockEditor` hook to address this?

Reply:
@youknowriad updated this in [352aa9e](https://github.com/WordPress/gutenberg/pull/60326/commits/352aa9e7db5d60da9f61add9cc7c048418417529). This removes a lot of the complexity because after testing a bit more it seems we can pretty much omit edits for user roles that cannot edit the template. 

This also fixes the request to use the view context as I mentioned

Reply:
That's better but I still wonder whether we should just reuse useEntityBlockEditor with an option? Wdyt?

Reply:
Hmm yeah, I hear your concerns. I'll have to see whether I can update the `useEntityBlockEditor` hook without adding too much complexity. It's not obvious to me how to best approach it right now but will experiment a bit.

Reply:
Okay I'm having a hard time modifying the `useEntityBlockEditor` hook. We can't call any of the hooks there conditionally but want to avoid doing a bunch of the work in case of the readonly mode. 

@youknowriad you let me know how you want to handle this. Do you want to push something to this branch? Do you want me to merge as is and then simplify it later? Open to whatever you think makes the most sense

Reply:
Let's merge as is. It doesn't introduce any API and address a need. We can consider the refactoring later if needed.